server.go

```diff
@@ -46,168 +46,238 @@ type Link struct {
-		for _, actor := range actors {
-			idx := -1
-			for i, node := range d3Resp.Nodes {
-				if actor == node.Title && node.Label == "actor" {
-					idx = i
-					break
+
+		d3Resp := D3Response{}
+		for result.Next() {
+			record := result.Record()
+			title, _ := record.Get("movie")
+			actors, _ := record.Get("cast")
+			d3Resp.Nodes = append(d3Resp.Nodes, Node{Title: title.(string), Label: "movie"})
+			movIdx := len(d3Resp.Nodes) - 1
+			for _, actor := range actors.([]interface{}) {
+				idx := -1
+				for i, node := range d3Resp.Nodes {
+					if actor == node.Title && node.Label == "actor" {
+						idx = i
+						break
+					}
+				}
+				if idx == -1 {
+					d3Resp.Nodes = append(d3Resp.Nodes, Node{Title: actor.(string), Label: "actor"})
+					d3Resp.Links = append(d3Resp.Links, Link{Source: len(d3Resp.Nodes) - 1, Target: movIdx})
+				} else {
+					d3Resp.Links = append(d3Resp.Links, Link{Source: idx, Target: movIdx})
 				}
-			}
-			if idx == -1 {
-				d3Resp.Nodes = append(d3Resp.Nodes, Node{Title: actor, Label: "actor"})
-				d3Resp.Links = append(d3Resp.Links, Link{Source: len(d3Resp.Nodes) - 1, Target: movIdx})
-			} else {
-				d3Resp.Links = append(d3Resp.Links, Link{Source: idx, Target: movIdx})
 			}
 		}
+
+		err = json.NewEncoder(w).Encode(d3Resp)
+		if err != nil {
+			log.Println("error writing graph response:", err)
+		}
 	}
+}
 
-	err = json.NewEncoder(w).Encode(d3Resp)
+func main() {
+	configuration := parseConfiguration()
+	driver, err := configuration.newDriver()
 	if err != nil {
-		log.Println("error writing graph response:", err)
+		log.Fatal(err)
 	}
+	defer unsafeClose(driver)
+	serveMux := http.NewServeMux()
+	serveMux.HandleFunc("/", defaultHandler)
+	serveMux.HandleFunc("/search", searchHandlerFunc(driver, configuration.Database))
+	serveMux.HandleFunc("/movie/", movieHandlerFunc(driver, configuration.Database))
+	serveMux.HandleFunc("/graph", graphHandler(driver, configuration.Database))
+
+	var port string
+	var found bool
+	if port, found = os.LookupEnv("PORT"); !found {
+		port = "8080"
+	}
+	panic(http.ListenAndServe(":"+port, httpgzip.NewHandler(serveMux)))
 }
 
-func init() {
-	if os.Getenv("GRAPHENEDB_URL") != "" {
-		neo4jURL = os.Getenv("GRAPHENEDB_URL")
+func internalServerError(w http.ResponseWriter, err error) {
+	w.WriteHeader(500)
+	w.Header().Set("Content-Type", "text/plain")
+	w.Write([]byte(fmt.Errorf("unexpected error: %w", err).Error()))
+}
+
+func parseLimit(req *http.Request) int {
+	limits := req.URL.Query()["limit"]
+	limit := 50
+	if len(limits) > 0 {
+		var err error
+		if limit, err = strconv.Atoi(limits[0]); err != nil {
+			limit = 50
+		}
 	}
+	return limit
 }
 
-func main() {
-	serveMux := http.NewServeMux()
-	serveMux.HandleFunc("/", defaultHandler)
-	serveMux.HandleFunc("/search", searchHandler)
-	serveMux.HandleFunc("/movie/", movieHandler)
-	serveMux.HandleFunc("/graph", graphHandler)
+func parseConfiguration() *Neo4jConfiguration {
+	return &Neo4jConfiguration{
```

Comment:
I just need to parameterize this properly

Reply:
Or maybe not, [some examples](https://github.com/neo4j-examples/movies-javascript-bolt/blob/2e8324c7754cee3a53bc6bce5684892925594033/src/neo4jApi.js#L7) hardcode the password as well ^^

Reply:
We should use env variables for the credentials and default perhaps to the demo.neo4jlabs.com one with movies/movies and db movies